lib/src/validator/dependency.dart

```diff
@@ -75,18 +75,20 @@ class DependencyValidator extends Validator {
           validateSdkConstraint(_firstGitPathVersion,
               "Older versions of pub don't support Git path dependencies.");
         }
-      } else if (constraint.isAny) {
-        _warnAboutNoConstraint(dependency);
-      } else if (constraint is Version) {
-        _warnAboutSingleVersionConstraint(dependency);
-      } else if (constraint is VersionRange) {
-        if (constraint.min == null) {
-          _warnAboutNoConstraintLowerBound(dependency);
-        } else if (constraint.max == null) {
-          _warnAboutNoConstraintUpperBound(dependency);
+      } else {
+        _warnAboutPrerelease(dependency);
+        if (constraint.isAny) {
+          _warnAboutNoConstraint(dependency);
+        } else if (constraint is Version) {
+          _warnAboutSingleVersionConstraint(dependency);
+        } else if (constraint is VersionRange) {
+          if (constraint.min == null) {
```

Comment:
Can't we move ` _warnAboutPrerelease(dependency);` down here? And call it as `_warnAboutPrerelease(dependency.name, constraint);` that way we don't need to do `is VersionRange` twice.

Reply:
I would like to also catch the `Version` case. (`Version` implements `VersionRange`) That's why I took it outside.

But I think I can restructure the logic to avoid the double check.